File: Data Structures & Algorithms/two-integer-sum/submission-8.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        // We use a dictionary to store each number as the key
        // and its index as the value
        Dictionary<int, int> nMap = new Dictionary<int, int>(); 

        for(int i = 0; i < nums.Length; i++) {
            // find the target
            int diff = target - nums[i];
            // check the map to see if the diff has been stored 
            if(nMap.ContainsKey(diff)) {
                return new int[] {nMap[diff], i}; 
            }
            nMap[nums[i]] = i;
        }

        return new int[] {}; 
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-8.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        // Set a pointer 
        int l = 1;
        for(int r = 1; r < nums.size(); r++) {
            if(nums[r - 1] != nums[r]) {
                nums[l++] = nums[r];
            }
        }
        return l;
    }
};